src/utility/image/image.cpp

```
@@ -0,0 +1,29 @@
+auto Image::details() const noexcept -> const Details& {
+    return pimpl->details;
+}
+
+auto Image::get_timestamp() const noexcept -> Clock::time_point {
+    return pimpl->timestamp;
+}
+auto Image::set_timestamp(Clock::time_point timestamp) noexcept -> void {
+    pimpl->timestamp = timestamp;
+}
+
+Image::Image() : pimpl{std::make_unique<Impl>()} {
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: pimpl

(cppcoreguidelines-pro-type-member-init)